google/resource-snippets/pubsub-v1/test_alternatives.sh

```diff
@@ -4,28 +4,30 @@ TF_PROJECT_ID={TF_PROJECT_ID}
 DM_PROJECT_ID={DM_PROJECT_ID}
 
 
-gcloud auth login
+gcloud auth application-default login
+export GOOGLE_APPLICATION_CREDENTIALS=~/.config/gcloud/application_default_credentials.json
 
 # Create DM resources
 gcloud config set project $DM_PROJECT_ID
 gcloud services enable deploymentmanager.googleapis.com
 gcloud deployment-manager deployments create d1 --config pubsub.yaml
 
 # Create Terraform resources
+pushd alternatives/tf
 gcloud config set project $TF_PROJECT_ID
-gcloud config list --format 'value(core.project)'
 terraform init
 terraform plan -var="deployment=d1" -var="project_id=${TF_PROJECT_ID}"
 terraform apply -auto-approve -var="deployment=d1" -var="project_id=${TF_PROJECT_ID}"
+popd
 
 
 # Export DM and TF resources for comparison
-gcloud pubsub subscriptions list --filter="labels.goog-dm:d1" --project $DM_PROJECT_ID | sed "s/${DM_PROJECT_ID}/PROJECT/" > dm.yaml
+gcloud pubsub subscriptions list --filter="labels.goog-dm:d1" --project $DM_PROJECT_ID | sed "s/${DM_PROJECT_ID}/PROJECT/" > /tmp/dm.yaml
 
-gcloud pubsub subscriptions list --filter="labels.goog-dm:d1" --project $TF_PROJECT_ID | sed "s/${TF_PROJECT_ID}/PROJECT/"  > tf.yaml
+gcloud pubsub subscriptions list --filter="labels.goog-dm:d1" --project $TF_PROJECT_ID | sed "s/${TF_PROJECT_ID}/PROJECT/"  > /tmp/tf.yaml
```

Comment:
So this example assumes a topic is attached to a subscription, because that's what happens in the TF/DM example. "gcloud pubsub subscriptions list" lists the subscription and its topic as well.

Reply:
I see. Never mind then, thanks for double-checking that!